src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,649 @@
+        try {
+            detailContainer = mapper.readValue(jsonResults, AmazonAccountDetailContainer.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return detailContainer.getAccountDetail();
+    }
+
+    protected boolean processAmazonAccountDetails(List<AmazonAccountDetail> amazonAccountDetails) {
+        boolean valid = true;
+        int count = 0;
+        for (AmazonAccountDetail amazonAccountDetail : amazonAccountDetails) {
+            if (StringUtils.equalsIgnoreCase(CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION, amazonAccountDetail.getKfsAccount())) {
+                LOG.info("processAmazonAccountDetails() Found an internal kfs account, ignoring this transaction for AWS account " 
+                    + amazonAccountDetail.getAwsAccount());
+            } else {
+                valid = processExternalBill(amazonAccountDetail) && valid;
+            }
+            count ++;
+        }
+        LOG.info("Processed " + count + " AWS records");
+        return valid;
+    }
+
+    private boolean processExternalBill(AmazonAccountDetail amazonAccountDetail) {
+        boolean valid = true;
+        try {
+            TransactionDTO transactionDTO = buildTransactionDTO(amazonAccountDetail);
+            if (transactionDTO.isFailingError()) {
+                valid = false;
+                LOG.error("processAmazonAccountDetails() there errors that prevent a creation of a DI " + amazonAccountDetail.toString());
+            } else if (shouldBuildDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO.getAmount())) {
+                buidAndRouteDistributionOfIncomeDocument(amazonAccountDetail, transactionDTO);
+            }
+        } catch (Throwable e) {
+            valid = false;
+            LOG.error("processAmazonAccountDetails() there was a problem building a DI for " + amazonAccountDetail.toString(), e);
+            GlobalVariables.getMessageMap().getErrorMessages().clear();
+        }
+        return valid;
+    }
+    
+    protected TransactionDTO buildTransactionDTO(AmazonAccountDetail amazonAccountDetail) {
+        TransactionDTO transactionDTO = new TransactionDTO();
+        transactionDTO.setLineDescription(buildAccountingLineDescription());
+        
+        transactionDTO.setAmount(convertCostStringToKualiDecimal(amazonAccountDetail.getCost()));
+        if (transactionDTO.getAmount().isLessEqual(KualiDecimal.ZERO)) {
+            transactionDTO.setFailingError(true);
+        }
+        
+        if (StringUtils.contains(amazonAccountDetail.getCostCenter(), KFSConstants.DASH)) {
+            //could be account & subaccount or whole transaction string
+            String[] costCenterArray = StringUtils.split(amazonAccountDetail.getCostCenter(), KFSConstants.DASH);
+            if (costCenterArray.length == 2) {
+                updateTransactionDTOAccountSubAccount(amazonAccountDetail, transactionDTO);
```

Comment:
I wonder if there is a way to make this behavior polymorphic and eliminate the if/else if. Maybe there could be sub-classes of the TransactionDTO that implement their own version of updateTransactionDTOAccountString or something?